### PR TITLE
[NO3] func_801BE080 func_801C13F8 func_801C9BC0

### PR DESCRIPTION
**Pull request by sozud**

### CODE REVIEW

File: src/st/no3/377D4.c [hunk 1]
```diff
@@ -175,7 +205,32 @@ INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801BD71C);
 
 INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801BDEB0);
 
-INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801BE080);
+extern Entity D_8007C0D8[];
```
Comment (Xeeynamo) [inline src/st/no3/377D4.c:208]: Can you move this one into `game.h`? Everything from `80010000` up to `800A0000` goes under `game.h`, which is the shared memory area across all the overlays. So it makes sense to not declare it exclusively inside the overlay code.

File: src/st/no3/377D4.c [hunk 2]
```diff
@@ -231,7 +286,25 @@ INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C11A4);
...
+    s32 var_s1;
+    s8 temp_s4;
+
+    temp_s4 = Random() & 3;
+    temp_s3 = ((Random() & 0xF) << 8) - 0x800;
+    for (var_s1 = 0; var_s1 < 6; var_s1++) {
+        temp_v0 = AllocEntity(D_8007D858, &D_8007D858[MaxEntityCount]);
+        if (temp_v0 != NULL) {
+            func_801C3F38(0x62, g_CurrentEntity, temp_v0);
+            temp_v0->subId = 2;
```
Comment (Xeeynamo) [inline src/st/no3/377D4.c:301]: `temp_v0` can be renamed into either `entity` or `newEntity`

File: src/st/no3/377D4.c [hunk 3]
```diff
@@ -231,7 +286,25 @@ INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C11A4);
...
-INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C13F8);
+void func_801C13F8() {
+    Entity* temp_v0;
+    s16 temp_s3;
+    s32 var_s1;
+    s8 temp_s4;
+
+    temp_s4 = Random() & 3;
+    temp_s3 = ((Random() & 0xF) << 8) - 0x800;
+    for (var_s1 = 0; var_s1 < 6; var_s1++) {
```
Comment (Xeeynamo) [inline src/st/no3/377D4.c:297]: `var_s1` can be renamed as `i`

File: src/st/no3/377D4.c [hunk 4]
```diff
@@ -929,7 +1002,22 @@ INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C9A2C);
...
 
-INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C9BC0);
+void func_801C9BC0(void) {
+    Entity* entity;
+    s8 temp_s4 = Random() & 3;
+    s16 temp_s3 = ((Random() & 0xF) << 8) - 0x800;
+    s32 i;
+
+    for (i = 0; i < 6; i++) {
+        entity = AllocEntity(D_8007D858, &D_8007D858[32]);
```
Comment (Xeeynamo) [inline src/st/no3/377D4.c:1012]: In other parts of the code, we use `MaxEntityCount` instead of `32`

**Review by Xeeynamo — CHANGES_REQUESTED** [review]
Thank you for the contribution! I requested some minor cosmetic changes before merging.

**Comment by sozud** [comment]
Thanks, the comments should be resolved now.

**Review by Xeeynamo — APPROVED** [review]

**Pull request merged**